server/src/main/java/no/fractal/socket/send/AbstractMessage.java

```diff
@@ -0,0 +1,150 @@
+package no.fractal.socket.send;
+
+import no.fractal.socket.StreamUtil;
+import no.fractal.util.StreamUtils;
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+
+import java.io.*;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+
+public abstract class AbstractMessage {
+
+    private final String messageType;
+    private final ArrayList<Segment> segments;
+
+
+    protected AbstractMessage(String messageType
+    ) {
+        this.messageType = messageType;
+        this.segments = new ArrayList<>();
+    }
+
+    protected void addSegment(Segment segment){
+        this.segments.add(segment);
+    }
+
+
+    public void writeToStream(BufferedOutputStream outputStream) throws IOException{
+
+        byte[] typeBytes = this.messageType.getBytes();
+        byte[] segmentBytes = getMainSegmentMeta().toString().getBytes();
+
+        // byte count
+        int byteCont = segments.stream().map(Segment::getBodyByteSize).reduce(Integer::sum).get();
+        byteCont += segmentBytes.length;
+        byteCont += typeBytes.length;
+        byteCont += 14; // hederstuff
```

Comment:
hva skjer her? hva dem 14 e?

Reply:
# TOTAL PAYLOAD SIZE
PAYLOAD_LENGTH = 4
# SIZE OF THE PAYLOAD IDENTIFIER
PAYLOAD_NAME_LENGTH = 2
# SIZE OF SEGMENTS HEADER
SEGMENTS_LENGTH = 4
# SIZE OF JSON PAYLOAD
JSON_LENGTH = 4

Reply:
wow den va markdown det ja
summa fra python filen sine numerstørrelse heddera

stemme vel og merke ikke siden json ikke e med der men hvis du har veldig lyst kan vi gott bare smett den inn